apps/dashr-ncbi-explorer/app.R

```diff
@@ -0,0 +1,654 @@
+appName <- Sys.getenv("DASH_APP_NAME")
+if (appName != "") {
+  pathPrefix <- sprintf("/%s/", appName)
+  
+  Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+             DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+}
+
+# set working directory
+setwd("/app")
+
+library(dash)
+library(dashBio)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(Biostrings)
+library(readr)
+library(ape)
+library(stringr)
+library(rentrez)
+library(dashTable)
+library(plotly)
+
+app <- Dash$new()
+
+# Functions and Data Loading
+
+FASTA_DATA <- readBStringSet(filepath = "data/alignment_viewer_p53_clustalo.FASTA")
+
+blank_dataframe <- data.frame(matrix(ncol=0,nrow=0))
+
+generate_table <- function(df, nrows=20)
+  
+  # function generates a dash table from a supplied data frame (df)
+  
+  #  and number of rows (nrows) to display
+{
+  
+  n <- min(nrows, nrow(df))
+  
+  rows <- lapply(seq(1, n), function(i) {
+    
+    htmlTr(children = lapply(as.character(df[i,]), htmlTd))
+    
+  })
+  
+  header <- htmlTr(children = lapply(names(df), htmlTh))
+  
+  htmlTable(
+    
+    children = c(list(header), rows)
+  )
+}
+
+# App Layout Elements
+
+#Header
+
+header <- htmlDiv(
+  id = "app-page-header",
+  style = list(
+    width = "100%",
+    background = "#262B3D" ,
+    color = "#E2EFFA"
+  ),
+  children = list(
+    htmlA(
+      id = "dashbio-logo",
+      children = list(
+        htmlImg(src='assets/plotly-dash-bio-logo.png', height = '36', width = '180',
+                style = list('top' = '10', 'margin' = '10px'))
+      ),
+      href = "/Portal"
+    ),
+    htmlH2("NCBI Nucleotide DB Explorer"),
+    htmlA(
+      id = "gh-link",
+      children = list("View on GitHub"),
+      href = "https://github.com/plotly/dash-sample-apps/tree/master/apps/dashr-ncbi-explorer",
+      style = list(color = "white", border = "solid 1px white")
+    ),
+    htmlImg(
+      src = "assets/GitHub-Mark-Light-64px.png"
+    )
+  )
+)
+
+#First Row 
+
+first_row <- htmlDiv(list(
+  htmlDiv(children = list(
+    htmlH3("Search for a Dataset:", style = list("margin" = 10)),
+    htmlP("Use keywords to search the Nucleotide database for a gene or organism of interest.
+                    This will retrieve the top 10 related datasets' Accession IDs along with their descriptions.
+                    ", style = list("margin" = 10)),
+    htmlP('Example searches: "Basiliscus basiliscus[Organism]" or "BRCA1[Gene]"
+                    ', style = list("margin" = 10)),
+    htmlP('Select the table cell containing your desired Accession ID to add this dataset
+    to the alignment and generate a new figure Then, filter through datasets to analyze the CpG patterns
+    and nucleotide compositions for your sequence of interest.
+                    ', style = list("margin" = 10)),
```

Comment:
could place `("margin" = 10)` in a css file since its referenced many times

Reply:
I think in this case, while margin is referenced a few times, but the styles of the individual containers are different regardless. A couple of the times it's also used as an argument for `plot_ly` figure layouts.